### PR TITLE
Switch CallSet to track RpoDigests instead of ProcedureIds

### PR DESCRIPTION
**Pull request by bobbinth**
Addresses #916.

### CODE REVIEW

File: assembly/src/assembler/context.rs [hunk 1]
```diff
@@ -335,8 +335,8 @@ impl ModuleContext {
 
     /// Returns a [Procedure] with the specified ID, or None if a compiled procedure with such ID
```
Comment (frisitano) [inline assembly/src/assembler/context.rs:336]: Would be good to update the comment to reflect the fact the lookup is now done against the mast root.

File: assembly/src/procedures/mod.rs [hunk 2]
```diff
@@ -64,18 +64,22 @@ impl Procedure {
...
-    /// Returns a root of this procedure's MAST.
-    pub fn code_root(&self) -> &CodeBlock {
+    /// Returns the root of this procedure's MAST.
+    pub fn mast_root(&self) -> RpoDigest {
+        self.code_root.hash()
+    }
+
+    /// Returns a reference to the MAST of this procedure.
+    pub fn code(&self) -> &CodeBlock {
         &self.code_root
```
Comment (frisitano) [inline assembly/src/procedures/mod.rs:78]: Should we consider renaming the `code_root` field to `code` such that we have parity with the accessor?

**Review by frisitano — APPROVED** [review]
Looks good.  Left a couple of comments.

File: assembly/src/procedures/mod.rs (hunk 2 above)
Comment (bobbinth) [reply assembly/src/procedures/mod.rs:78]: Agreed! I've actually fixed this in one of the subsequent PRs.

**Pull request merged**